1/ToDoApp/src/components/AppLogin/AppLogin.vue

```diff
@@ -0,0 +1,62 @@
+<template>
+  <div>
+    <h4 class='mv3'>{{login ? 'Login' : 'Sign Up'}}</h4>
+    <div class='flex flex-column'>
+      <input
+        v-show="!login"
+        v-model="name"
+        type="text"
+        placeholder="Your name">
+      <input
+        v-model="email"
+        type="text"
+        placeholder="Your email address">
+      <input
+        v-model="password"
+        type="password"
+        placeholder="Password">
+    </div>
+    <div class='flex mt3'>
+   <!--
+      <b-button variant="success"
+        class='pointer mr2 button'
+        @click="confirm()">
+        {{login ? 'login' : 'create account'}}
+      </b-button>
+    -->
+      <div
+        class='pointer button'
+        @click="login = !login">
+        {{login ? 'need to create an account?' : 'already have an account?'}}
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+
+ // import { GC_USER_ID, GC_AUTH_TOKEN } from '../../constants/settings'
+
+
+  export default {
+    name: 'AppLogin',
+    data () {
+      return {
+        email: '',
+        login: true,
+        name: '',
+        password: ''
+      }
+    },
+    methods: {
+      confirm () {
+        // ... you'll implement this in a bit
+      },
+      saveUserData (id, token) {
+        localStorage.setItem(GC_USER_ID, id)
+        localStorage.setItem(GC_AUTH_TOKEN, token)
```

Comment:
Once we dive into NuxtJS, you will discover [nuxt apollo module](https://github.com/nuxt-community/apollo-module) which comes with [login helpers](https://github.com/nuxt-community/apollo-module#authentication). So, you don't have to implement it yourself.